lib/xray/middleware.rb

```diff
@@ -59,7 +59,7 @@ def call(env)
           # Modifying the original response obj maintains compatibility with other middlewares
           if ActionDispatch::Response === response
             response.body = [body]
-            response.header['Content-Length'] = content_length
+            response.header['Content-Length'] = content_length unless response.try(:committed?)
```

Comment:
Yeah, I'll open on an issue in the future, no problem. Just wanted to make the remark in context.

Thanks for looking into it!